__tests__/app.test.js

```diff
@@ -32,3 +32,17 @@ describe("GET api/nonsense ", () => {
       });
   });
 });
+describe("GET /api", () => {
+  test("should respond with an object describing all the available endpoints ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        const { allApi } = body;
+        const arrOfKeys = Object.keys(allApi);
+        arrOfKeys.forEach((key) => {
+          expect(typeof key).toBe("string");
+        });
```

Comment:
Every key on an object is a string, you could be more specific with your test, you know exactly what the response should look like.

Reply:
The problem I have that response should be dynamic. So don't want to use toEqual. And really don't know how to make it dynamic.